federation/pkg/federatedtypes/hpa.go

```diff
@@ -0,0 +1,735 @@
+	return a.client.AutoscalingV1().HorizontalPodAutoscalers(namespacedName.Namespace).Get(namespacedName.Name, metav1.GetOptions{})
+}
+
+func (a *HpaAdapter) FedList(namespace string, options metav1.ListOptions) (pkgruntime.Object, error) {
+	return a.client.AutoscalingV1().HorizontalPodAutoscalers(namespace).List(options)
+}
+
+func (a *HpaAdapter) FedUpdate(obj pkgruntime.Object) (pkgruntime.Object, error) {
+	secret := obj.(*autoscalingv1.HorizontalPodAutoscaler)
+	return a.client.AutoscalingV1().HorizontalPodAutoscalers(secret.Namespace).Update(secret)
+}
+
+func (a *HpaAdapter) FedWatch(namespace string, options metav1.ListOptions) (watch.Interface, error) {
+	return a.client.AutoscalingV1().HorizontalPodAutoscalers(namespace).Watch(options)
+}
+
+func (a *HpaAdapter) ClusterCreate(client kubeclientset.Interface, obj pkgruntime.Object) (pkgruntime.Object, error) {
+	secret := obj.(*autoscalingv1.HorizontalPodAutoscaler)
+	return client.AutoscalingV1().HorizontalPodAutoscalers(secret.Namespace).Create(secret)
+}
+
+func (a *HpaAdapter) ClusterDelete(client kubeclientset.Interface, nsName types.NamespacedName, options *metav1.DeleteOptions) error {
+	return client.AutoscalingV1().HorizontalPodAutoscalers(nsName.Namespace).Delete(nsName.Name, options)
+}
+
+func (a *HpaAdapter) ClusterGet(client kubeclientset.Interface, namespacedName types.NamespacedName) (pkgruntime.Object, error) {
+	return client.AutoscalingV1().HorizontalPodAutoscalers(namespacedName.Namespace).Get(namespacedName.Name, metav1.GetOptions{})
+}
+
+func (a *HpaAdapter) ClusterList(client kubeclientset.Interface, namespace string, options metav1.ListOptions) (pkgruntime.Object, error) {
+	return client.AutoscalingV1().HorizontalPodAutoscalers(namespace).List(options)
+}
+
+func (a *HpaAdapter) ClusterUpdate(client kubeclientset.Interface, obj pkgruntime.Object) (pkgruntime.Object, error) {
+	secret := obj.(*autoscalingv1.HorizontalPodAutoscaler)
+	return client.AutoscalingV1().HorizontalPodAutoscalers(secret.Namespace).Update(secret)
+}
+
+func (a *HpaAdapter) ClusterWatch(client kubeclientset.Interface, namespace string, options metav1.ListOptions) (watch.Interface, error) {
+	return client.AutoscalingV1().HorizontalPodAutoscalers(namespace).Watch(options)
+}
+
+func (a *HpaAdapter) NewTestObject(namespace string) pkgruntime.Object {
+	return &autoscalingv1.HorizontalPodAutoscaler{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-hpa-",
+			Namespace:    namespace,
+		},
+		Spec: autoscalingv1.HorizontalPodAutoscalerSpec{
+			ScaleTargetRef: autoscalingv1.CrossVersionObjectReference{
+				Kind: "HorizontalPodAutoscaler",
+				Name: "myhpa",
+			},
+			MaxReplicas: int32(10),
+		},
+	}
+}
+
+func (a *HpaAdapter) IsScheduler() bool {
+	return true
+}
+
+func (a *HpaAdapter) Schedule(obj pkgruntime.Object, current map[string]pkgruntime.Object) (map[string]pkgruntime.Object, error) {
```

Comment:
This method is pushing 400 lines.  Not only is that fundamentally untestable, it's also impossible to review effectively.   I'd like to see you refine your ideas before I commit to providing feedback, because in their current form I don't think the code is capable of clearly communicating your intent.